engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -116,6 +128,14 @@ public void activate(final long key, final JobRecord record) {
     deadlinesColumnFamily.put(deadlineJobKey, DbNil.INSTANCE);
   }
 
+  @Override
+  public void recurAfterBackoff(final long key, final JobRecord record) {
+    updateJob(key, record, State.ACTIVATABLE);
+    jobTypeKey.wrapBuffer(record.getTypeBuffer());
```

Comment:
:x: We need to wrap the job key instead of the job type.

Same as the other comment on `fail()`, I made a wrong suggestion. I'm sorry.

```suggestion
    jobKey.wrapLong(key);
```